python_modules/libraries/dagster-dg/dagster_dg/cli/list.py

```diff
@@ -267,44 +294,47 @@ def list_defs_command(output_json: bool, path: Path, **global_options: object) -
     cli_config = normalize_cli_config(global_options, click.get_current_context())
     dg_context = DgContext.for_project_environment(path, cli_config)
 
-    result = dg_context.external_components_command(
-        [
-            "list",
-            "definitions",
-            "-m",
-            dg_context.code_location_target_module_name,
-        ],
-        # Sets the "--location" option for "dagster-components definitions list"
-        # using the click auto-envvar prefix for backwards compatibility on older versions
-        # before that option was added
-        additional_env={"DG_CLI_LIST_DEFINITIONS_LOCATION": dg_context.code_location_name},
-    )
+    # On newer versions, we use a dedicated channel in the form of a tempfile that will _only_ have
+    # the expected output written to it.
+    if (
+        dg_context.dagster_version
+        >= MIN_DAGSTER_COMPONENTS_LIST_DEFINITIONS_OUTPUT_FILE_OPTION_VERSION
```

Comment:
do you need to add an editable check here? or are you relying on the dev version always being >= every published version?